projects/RabbitMQ.Client/client/events/AsyncEventingBasicConsumer.cs

```diff
@@ -45,12 +45,14 @@ public override async Task HandleBasicConsumeOk(string consumerTag)
         }
 
         ///<summary>Fires the Received event.</summary>
-        public override async Task HandleBasicDeliver(string consumerTag, ulong deliveryTag, bool redelivered, string exchange, string routingKey, IBasicProperties properties, ReadOnlyMemory<byte> body)
+        public override Task HandleBasicDeliver(string consumerTag, ulong deliveryTag, bool redelivered, string exchange, string routingKey, IBasicProperties properties, ReadOnlyMemory<byte> body)
         {
-            await base.HandleBasicDeliver(consumerTag, deliveryTag, redelivered, exchange, routingKey, properties, body).ConfigureAwait(false);
-            await (Received?.Invoke(
-                this,
-                new BasicDeliverEventArgs(consumerTag, deliveryTag, redelivered, exchange, routingKey, properties, body)) ?? Task.CompletedTask).ConfigureAwait(false);
+            if (Received != null)
```

Comment:
This change raises red flags for me and does not seem to be related to the rest of the changes in this PR. All of the other methods in `AsyncEventingBasicConsumer` are `async`, and use `await` to call the base class methods (which, strangely enough, don't do anything).

Reply:
The reason for this change is that the base.HandleBasicDeliver is a no-op, so it's removed. That leaves the method with just one 'await' call and no other operation that takes place. Therefore there is no need to create async state machine and we can just return the task directly and save ourselves some CPU cycles.

However I can see that this might seem strange, and out-of-line with the rest of the code. It's a micro-optimization, which at this point is probably not necessary. I'll change it back.

Reply:
I will make a note to review this code for 7.0, thanks.